system/jlib/jsocket.cpp

```diff
@@ -1161,6 +1167,11 @@ int CSocket::peer_name(char *retname,size32_t namemax)
 
 SocketEndpoint &CSocket::getPeerEndpoint(SocketEndpoint &ep)
 {
+    if (peerEpSet)
+    {
+        ep = peerEp;
```

Comment:
technically I suspect this isn't well defined because peerEP isn't atomic - so the compiler is theoretically allowed to read peerEp before testing peerEpSet.  It would also be legal for the compiler to reorder the store to peerEpSet before setting peerEp (and some processors may reorder them).  In practice I suspect it is highly unlikely to cause any problems.
I think the cleanest "correct" solution is to read the value of peerEp within a critical section, but the test for peerEpSet probably needs to be inside the crit sec as well.

Reply:
I agree, good find.  Put this inside a CS